Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -1212,129 +968,123 @@ private bool MatchesPublishedServerUrl(IPObject source, bool isInExternalSubnet,
+            }
+            else
+            {
+                _logger.LogDebug("{Source}: Matching bind address override found: {Address}", source, bindPreference);
+            }
+
             return true;
         }
 
         /// <summary>
-        /// Attempts to match the source against a user defined bind interface.
+        /// Attempts to match the source against the user defined bind interfaces.
         /// </summary>
         /// <param name="source">IP source address to use.</param>
         /// <param name="isInExternalSubnet">True if the source is in the external subnet.</param>
         /// <param name="result">The result, if a match is found.</param>
         /// <returns><c>true</c> if a match is found, <c>false</c> otherwise.</returns>
-        private bool MatchesBindInterface(IPObject source, bool isInExternalSubnet, out string result)
+        private bool MatchesBindInterface(IPAddress source, bool isInExternalSubnet, out string result)
         {
             result = string.Empty;
-            var addresses = _bindAddresses.Exclude(_bindExclusions, false);
 
-            int count = addresses.Count;
-            if (count == 1 && (_bindAddresses[0].Equals(IPAddress.Any) || _bindAddresses[0].Equals(IPAddress.IPv6Any)))
+            int count = _interfaces.Count;
+            if (count == 1 && (_interfaces[0].Equals(IPAddress.Any) || _interfaces[0].Equals(IPAddress.IPv6Any)))
             {
                 // Ignore IPAny addresses.
                 count = 0;
             }
 
-            if (count != 0)
+            if (count == 0)
             {
-                // Check to see if any of the bind interfaces are in the same subnet.
-
-                IPAddress? defaultGateway = null;
-                IPAddress? bindAddress = null;
+                return false;
+            }
 
-                if (isInExternalSubnet)
+            IPAddress? bindAddress = null;
+            if (isInExternalSubnet)
+            {
+                var externalInterfaces = _interfaces.Where(x => !IsInLocalNetwork(x.Address))
+                    .OrderBy(x => x.Index)
+                    .ToList();
+                if (externalInterfaces.Count > 0)
                 {
-                    // Find all external bind addresses. Store the default gateway, but check to see if there is a better match first.
-                    foreach (var addr in addresses.OrderBy(p => p.Tag))
+                    // Check to see if any of the external bind interfaces are in the same subnet as the source.
+                    // If none exists, this will select the first external interface if there is one.
+                    bindAddress = externalInterfaces
+                        .OrderByDescending(x => x.Subnet.Contains(source))
+                        .ThenBy(x => x.Index)
+                        .Select(x => x.Address)
+                        .FirstOrDefault();
```

Comment:
Default will never be returned as you have already verified that the list contains at least one element